### PR TITLE
Move plugin installer settings to new paths

### PR DESCRIPTION
**Pull request by borysiasty**
Move all plugin installer QSetttings paths from the depreciated **[Qgis]** branch to a new top-level **[PluginInstaller]** (except dialog geometry that goes directly to **[Windows]**).

| old                                    | new                                             |
 --------------------------------------- | -----------------------------------------------
/Qgis/plugin-installer                   | PluginInstaller
/Qgis/plugin-repos                       | PluginInstaller/repositories
/Qgis/plugin-seen                        | PluginInstaller/seenPlugins
/Qgis/plugin-installer/fetching_geometry | Windows/PluginInstallerFetchingDialog/geometry

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
@nathanw2 can you guide @borysiasty in adding settings translators for this?

**Comment by borysiasty** [comment]
@NathanW2 @rduivenvoorde - is it ok? Please merge if you don't have any objections. 
Btw. I removed the starting slashes - hopefully it's ok for all OSes.

**Comment by borysiasty** [comment]
@nyalldawson @NathanW2 should I just add the mapping to the 2to3migration.txt file ?

**Comment by NathanW2** [comment]
@borysiasty yep that looks good. If it works for you just dump the version.

File: python/pyplugin_installer/qgsplugininstallerfetchingdialog.py [hunk 1]
```diff
@@ -55,11 +55,11 @@ def __init__(self, parent):
         repositories.anythingChanged.connect(self.displayState)
 
         settings = QgsSettings()
-        self.restoreGeometry(settings.value("/Qgis/plugin-installer/fetching_geometry", QByteArray()))
+        self.restoreGeometry(settings.value("Windows/PluginInstallerFetchingDialog/geometry", QByteArray()))
```
Comment (elpaso) [inline python/pyplugin_installer/qgsplugininstallerfetchingdialog.py]: Would you consider using a `QgsSettins::Section` here? 
We can add more sections if needed: http://qgis.org/api/classQgsSettings.html#aba2f174ee000905cac8aff26665ed7c7

Comment (borysiasty) [reply python/pyplugin_installer/qgsplugininstallerfetchingdialog.py]: @elpaso it's just one of over hundred entries (mostly geometries) in the **[Windows]** group. There are also dozens of entries in the **[UI]** group. At the beginning [Windows] was designed for geometries, and [UI] for other settings I guess, but now they are a little mixed. 

Do we want to port both groups to the new `QgsSettings::Section::Gui` (for now only QgsLocator uses it) or to keep a separate group for windows geometries? Unless you have a strong opinion, I'd leave it in the [Windows] for now, so later we can grep them all and port to a proper QgsSettings::Section in one move.

Comment (elpaso) [reply python/pyplugin_installer/qgsplugininstallerfetchingdialog.py]: No strong opinions, here, I just would like to see qgis settings slowly but steadily moving to something more stuctured and ordered, to make is easier for us developers and for users to understand in which section a particular setting can be found, the current situation is a bit messy. 
I think we need a `Windows` section in `QgsSettings`, there is already a `Gui` that was meant for GUI settings (for instance to remember the last opened tab in a dialog etc.).

Comment (borysiasty) [reply python/pyplugin_installer/qgsplugininstallerfetchingdialog.py]: Ok, so I'll add the `Windows` section and port Installer and Manager there later today.

**Comment by borysiasty** [comment]
Closing for now, better paths are coming.

File: python/pyplugin_installer/qgsplugininstallerfetchingdialog.py (hunk 1 above)
Comment (borysiasty) [reply python/pyplugin_installer/qgsplugininstallerfetchingdialog.py]: @elpaso finally I didn't add the Section::Windows, as I realized the related settings were dead anyway.
Let's do it later in one move if/when porting existing geometry keys.

**Pull request closed without merging**